examples/dlint/diagnostics.rs

```diff
@@ -9,22 +9,45 @@ pub fn display_diagnostics(
   diagnostics: &[LintDiagnostic],
   source_file: &SourceTextInfo,
   filename: &str,
+  format: Option<&str>,
+) {
+  for diagnostic in diagnostics {
+    match format {
```

Comment:
There can't be a situation where one diagnostic is formatted using "compat" and other using "pretty" formatting. This function should be refactored to forward `diagnostics` slice to either `print_compat` or `print_pretty` helpers

Reply:
made some updates. Hope I understood all your comments correctly